test/contrib/test_autoguide.py

```diff
@@ -83,3 +84,71 @@ def body_fn(i, val):
     # test .sample_posterior method
     posterior_samples = guide.sample_posterior(random.PRNGKey(1), opt_state, sample_shape=(1000,))
     assert_allclose(np.mean(posterior_samples['coefs'], 0), true_coefs, rtol=0.1)
+
+
+def test_uniform_normal():
+    true_coef = 0.9
+    data = true_coef + random.normal(random.PRNGKey(0), (1000,))
+
+    def model(data):
+        alpha = sample('alpha', dist.Uniform(0, 1))
+        loc = sample('loc', dist.Uniform(0, alpha))
+        sample('obs', dist.Normal(loc, 0.1), obs=data)
+
+    opt_init, opt_update, get_params = optimizers.adam(0.01)
+    rng_guide, rng_init, rng_train = random.split(random.PRNGKey(1), 3)
+    guide = AutoDiagonalNormal(rng_guide, model, get_params)
+    svi_init, svi_update, _ = svi(model, guide, elbo, opt_init, opt_update, get_params)
+    opt_state, constrain_fn = svi_init(rng_init, model_args=(data,), guide_args=(data,))
+
+    def body_fn(i, val):
+        opt_state_, rng_ = val
+        loss, opt_state_, rng_ = svi_update(i, rng_, opt_state_, model_args=(data,), guide_args=(data,))
+        return opt_state_, rng_
+
+    opt_state, _ = fori_loop(0, 1000, body_fn, (opt_state, rng_train))
+    median = guide.median(opt_state)
+    assert_allclose(median['loc'], true_coef, rtol=0.05)
+    # test .quantile method
+    median = guide.quantiles(opt_state, [0.2, 0.5])
+    assert_allclose(median['loc'][1], true_coef, rtol=0.1)
+
+
+def test_dynamic_supports():
```

Comment:
This test seems to pass without the `if is_autoguide` block in `svi.py`. Could you modify this so that it fails? It will be nice to be able to refactor that part, and have a unit test tell us that the refactoring handles the case of dynamic support for autoguides.

Reply:
@neerajprad do you mean by replacing `if is_autoguide` by `if False`, the test will pass?

Reply:
> @neerajprad do you mean by replacing if is_autoguide by if False, the test will pass? 

Yes, the test passes without this block.

> Do you modify skip_dist_transforms=is_autoguide too?

I didn't modify this. I was hoping that changing either would result in a failing test.

Reply:
In my system, setting `if False` will throw. Maybe it is precision issue, I'll check it again.
```
>       assert_allclose(actual_loss, expected_loss)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatch: 100%
E       Max absolute difference: 277.75
E       Max relative difference: 0.00472562
E        x: array(58497.617, dtype=float32)
E        y: array(58775.367, dtype=float32)
```

Reply:
Oh..sorry my bad! I have been running the test in `test_svi.py` instead of `test_autoguide.py`, which will obviously pass. Sorry for wasting your time.

Reply:
I am able to replicate the failure now; thanks for checking.

Reply:
Hmm, I run `JAX_ENABLE_X64=1 XLA_FLAGS=--xla_cpu_enable_fast_math=false pytest -vs test/contrib/test_autoguide.py -k dynamic`, the test failed with a large margin
```
>       assert_allclose(actual_loss, expected_loss)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatch: 100%
E       Max absolute difference: 37648.07914654
E       Max relative difference: 0.44394272
E        x: array(47155.831707)
E        y: array(84803.910853)
```
I think it is just lucky to pass in your system. :D I have revised the test to have a larger relative difference (~0.01 in my system), hope that it will fail in your case.

Reply:
Haha, np!